File: csv.go
package main

import (
	"github.com/dfirebaugh/mdsrs/models"
	"github.com/dfirebaugh/mdsrs/store"
	"github.com/sirupsen/logrus"
)


type CSVService struct {
	decks map[string]*models.Deck
}

func (a *CSVService) ExportDeck(deckName string) string {
	deck, ok := a.decks[deckName]
	if !ok {
		logrus.Errorf("deck not found: %s", deckName)
		return ""
	}

	csvData, err := store.ExportDeckToCSV(deck)
	if err != nil {
		logrus.Errorf("failed to export deck to CSV: %v", err)
		return ""
	}

	return csvData
}

func (a *CSVService) ImportDeck(deckName string, csvData string) error {
	deck, err := store.ImportCSVDeck(csvData, deckName)
	if err != nil {
		return err
	}

	a.decks[deck.Name] = deck
	return nil
}

File: frontend/src/services/csv.js
import { ImportDeck, ExportDeck } from "../../wailsjs/go/main/CSVService.js";

export default class CSVService {
	/**
	 * Import a deck from CSV data.
	 * @param {string} deckName - The name for the new deck.
	 * @param {string} csvData - The CSV data as a string.
	 * @returns {Promise<void>}
	 */
	static Import(deckName, csvData) {
		return ImportDeck(deckName, csvData)
	}

	/**
	 * Export a deck to a CSV file.
	 * @param {string} deckName - The deck identifier.
	 * @returns {Promise<string>} The CSV data as a string, or empty string if error.
	 */
	static Export(deckName) {
		return ExportDeck(deckName)
	}
}
